lib/webvalve/manager.rb

```diff
@@ -19,16 +23,46 @@ def allow_url(url)
     end
 
     def setup
-      fake_service_configs.each do |config|
-        if config.should_intercept?
-          webmock_service config
-        else
-          allowlist_service config
+      return if disabled?
+
+      if intercepting?
+        fake_service_configs.each do |config|
+          if !WebValve.env.test? && config.explicitly_enabled?
+            allowlist_service config
+          else
+            webmock_service config
+          end
         end
+        WebMock.disable_net_connect! webmock_disable_options
+        WebMock.enable!
       end
 
-      WebMock.enable!
-      WebMock.disable_net_connect! webmock_disable_options
+      if allowing?
+        fake_service_configs.each do |config|
+          if config.explicitly_disabled?
+            webmock_service config
+          end
+        end
+        if fake_service_configs.any?(&:explicitly_disabled?)
+          WebMock.allow_net_connect!
+          WebMock.enable!
+        end
+      end
+    end
+
+    # @api private
+    def disabled?
+      !intercepting? && !allowing?
+    end
+
+    # @api private
+    def intercepting?
+      in_always_intercepting_env? || ENABLED_VALUES.include?(ENV['WEBVALVE_ENABLED'])
+    end
+
+    # @api private
+    def allowing?
+      !in_always_intercepting_env? && DISABLED_VALUES.include?(ENV['WEBVALVE_ENABLED'])
```

Comment:
The more I look at this the more I want to rename this env variable to WEBVALVE_MODE and have it behave accordingly:
- unset means off, webvalve doesn't even load
- 'allowing', webvalve loads and defaults to allowing internet connections
- 'intercepting', webvalve loads and defaults to disallowing internet connections

But, uh, I dunno. Gonna make a decision this week and see how it goes in real codebases with real use cases.

Reply:
Unset vs Allowing feels conceptually very tricky, because in production they should behave identically (like, regardless of underlying stuff getting loaded or not). Like, I get where you're going with it, but unless you make a 3rd option called 'off' and make it explicitly togglable (like `WEBVALVE_MODE=off`) there's the potential for some surprises.

I still sort of stand by the logic that if you don't want this gem "on" in production (in the sense that it is loaded at runtime) then you should put it in the test/dev groups of the Gemfile like other non-prod dependencies. And that should be the README's default recommendation. Most devs get that soon as you put a gem like that in the production group, all bets are off, and I think that's okay...

Reply:
What if the options are
1. `WEBVALVE_ENABLED=1` intercepts all services by default and `FAKE_[SERVICE]_ENABLED=0` selectively toggles the fakes off (enabling the real services), otherwise
2. `WEBVALVE_ENABLED=0` (or, simply, unset) intercepts no services, and then `FAKE_[SERVICE]_ENABLED=1` can be used to intercept select services.

WEBVALVE_ENABLED ends up meaning more "default all services to enabled"

Reply:
WEBVALVE_ENABLED can default to 1 in dev/test -- and then prod-like envs that we want to totally isolate from real services can also set to 1 in prod.

Reply:
(I see that you wrote an ADR on this so feel free to take or leave these comments -- I'm just trying to get these thoughts down while I have them.)

Reply:
i actually wrote the ADR so that we could have this conversation a bit more concretely, so keep the thoughts coming 😄 

i tried to make my peace with the unset vs 0 meaning different things problem by saying that it would only impact folks trying to use the more advanced behavior that the tools supports.

Reply:
> 2\. `WEBVALVE_ENABLED=0` (or, simply, unset) intercepts no services, and then `FAKE_[SERVICE]_ENABLED=1` can be used to intercept select services.

the concern i have with this approach is that i wanted enabling webvalve in a production-like environment to require a sort of 2 keys strategy. i.e. i want you to have to say "i'd like to use webvalve" and "i'd like webalve to enable fake this service"

the main motivation for that, for me, is safety. i don't know if it's wise to allow someone to enable a fake service with just the one toggle.

Reply:
Maybe let's ditch `WEBVALVE_ENABLED` as an overloaded concept and go back to `WEBVALVE_ENABLED=0` meaning it's fully off, nothing activated, and `WEBVALVE_ENABLED=1` means it loads and enables RAILS_ENV-specific default behaviors. 

In dev/test this means enabling all fakes, but in production the two key process is this:

"i'd like to use webvalve"

> `WEBVALVE_ENABLED=1`

"i'd like webvalve to enable fakes in this service"

> `WEBVALVE_SERVICE_ENABLED_DEFAULT=0`
> or
> `[SERVICE]_ENABLED=0`

Reply:
And then, optionally, we can eventually solve the confusion around "[service]_enabled" meaning the _real_ service, not the _fake_ service, by renaming these things and inverting their meaning:

> `WEBVALVE_FAKE_SERVICE_ENABLED_DEFAULT=1`
> `FAKE_[SERVICE]_ENABLED=1`

And we're okay with that first thing being super verbose because it's only for power users who have read all the docs 😅

Reply:
implemented!